extensions/email/src/main/java/org/mule/extension/email/api/retriever/RetrieverConnection.java

```diff
@@ -0,0 +1,185 @@
+     *
+     * @param protocol the desired protocol to use. One of imap or pop3 (and its secure versions)
+     * @param username the username to establish the connection.
+     * @param password the password corresponding to the {@code username}.
+     * @param host the host name of the mail server
+     * @param port the port number of the mail server.
+     * @param connectionTimeout the socket connection timeout
+     * @param readTimeout the socket read timeout
+     * @param writeTimeout the socket write timeout
+     * @param properties additional custom properties.
+     * @param folder the folder to be opened in order to list emails.
+     * @param tlsContextFactory the tls context factory for creating the context to secure the connection
+     */
+    public RetrieverConnection(String protocol,
+                               String username,
+                               String password,
+                               String host,
+                               String port,
+                               long connectionTimeout,
+                               long readTimeout,
+                               long writeTimeout,
+                               Map<String, String> properties,
+                               String folder,
+                               TlsContextFactory tlsContextFactory) throws EmailConnectionException
+    {
+        super(protocol, username, password, host, port, connectionTimeout, readTimeout, writeTimeout, properties, tlsContextFactory);
+        try
+        {
+            this.store = session.getStore(protocol);
+            this.store.connect(username, password);
+            this.folder = store.getFolder(folder);
+        }
+        catch (MessagingException e)
+        {
+            throw new EmailConnectionException("Error while acquiring connection with the " + protocol + "store", e);
+        }
+    }
+
+    /**
+     * Creates a new instance of the of the {@link RetrieverConnection}.
+     *
+     * @param protocol the desired protocol to use. One of imap or pop3 (and its secure versions)
+     * @param username the username to establish the connection.
+     * @param password the password corresponding to the {@code username}.
+     * @param host the host name of the mail server
+     * @param port the port number of the mail server.
+     * @param connectionTimeout the socket connection timeout
+     * @param readTimeout the socket read timeout
+     * @param writeTimeout the socket write timeout
+     * @param properties additional custom properties.
+     * @param folder the folder to be opened in order to list emails.
+     */
+    public RetrieverConnection(String protocol,
+                               String username,
+                               String password,
+                               String host,
+                               String port,
+                               long connectionTimeout,
```

Comment:
in which unit are these timeouts? Consider using the org.mule.runtime.core.time.Time class